openmetadata-service/src/main/java/org/openmetadata/service/search/SearchClient.java

```diff
@@ -108,6 +108,8 @@ public interface SearchClient {
 
   void createAliases(IndexMapping indexMapping);
 
+  void addIndexAlias(IndexMapping indexMapping, String aliasName);
```

Comment:
As part of the `searchRepository.createIndex()` we will add the alias defined in the `indexMappin.json` -- see below. Would it be worth adding maybe a `dataInsightAlias` key to that file and the corresponding object so that as part of `searchRepository.createIndex()` we add the corresponding DI alias. I am thinking it would allow to scale adding new index to the DI more easily and limit some code duplication -- WDYT?

```
  public void createIndex(IndexMapping indexMapping) {
    try {
      if (!indexExists(indexMapping)) {
        String indexMappingContent = getIndexMapping(indexMapping);
        searchClient.createIndex(indexMapping, indexMappingContent);
        searchClient.createAliases(indexMapping);
      } 
```

Reply:
I'm not sure how this will work, whenever we add a new entry to `indexMappin.json`  it will try to first create a index and then try to create alias, in our case the index should already exist and we want to create alias for multiple indexes as part of DI app so not sure how it will work 🤔.

Also I feel the current logic is scalable as we only have to add the entity [type here](https://github.com/open-metadata/OpenMetadata/blob/bab93a84d269808466e326d45970ae47f984a38f/openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/DataInsightsApp.java#L81) (which we would have to do anyways if we plan to add a new index to DI) as soon as we add that the alias will get created automatically.

and I see the code duplication I think I can take care of that 🤔

Reply:
Sounds good.